--- uart16550.origin/rtl/verilog/uart_wb.v	2013-09-27 21:10:21.276063490 +0200
+++ uart16550/rtl/verilog/uart_wb.v	2013-09-27 21:09:58.788346170 +0200
@@ -214,11 +214,16 @@
 				wbstate <= 2;
 				wre <= 0;
 			end
-			2,3: begin
+			2: begin
 				wb_ack_o <= 0;
-				wbstate <= 0;
+				wbstate <= 3;
 				wre <= 0;
 			end
+			3: begin
+				wb_ack_o <= 0;
+				wbstate <= 0;
+				wre <= 1;
+			end
 		endcase
 
 assign we_o =  wb_we_is & wb_stb_is & wb_cyc_is & wre ; //WE for registers	
