Index: jtag_vpi/jtag_vpi.c
===================================================================
--- jtag_vpi.orig/jtag_vpi.c	2013-09-08 21:37:15.000000000 +0100
+++ jtag_vpi/jtag_vpi.c	2014-01-10 18:49:52.900534389 +0000
@@ -29,7 +29,7 @@
 
 #include <vpi_user.h>
 
-#define RSP_SERVER_PORT	50020
+#define RSP_SERVER_PORT	5555
 #define	XFERT_MAX_SIZE	512
 
 const char * cmd_to_string[] = {"CMD_RESET",
