--- adv_debug_sys.orig/rtl/verilog/adbg_crc32.v	2013-09-23 22:31:26.768176104 +0200
+++ adv_debug_sys/rtl/verilog/adbg_crc32.v	2013-09-23 22:45:16.236749331 +0200
@@ -63,7 +63,13 @@
 
 reg    [31:0] crc;
 wire   [31:0] new_crc;
+wire          data_sim;
 
+`ifdef SIM
+assign data_sim = (data === 1'bx || data === 1'b0) ? 1'b0 : 1'b1;
+`else
+assign data_sim = data;
+`endif
 
 // You may notice that the 'poly' in this implementation is backwards.
 // This is because the shift is also 'backwards', so that the data can
@@ -73,33 +79,33 @@
 assign new_crc[2] = crc[3];
 assign new_crc[3] = crc[4];
 assign new_crc[4] = crc[5];
-assign new_crc[5] = crc[6] ^ data ^ crc[0];
+assign new_crc[5] = crc[6] ^ data_sim ^ crc[0];
 assign new_crc[6] = crc[7];
 assign new_crc[7] = crc[8];
-assign new_crc[8] = crc[9] ^ data ^ crc[0];
-assign new_crc[9] = crc[10] ^ data ^ crc[0];
+assign new_crc[8] = crc[9] ^ data_sim ^ crc[0];
+assign new_crc[9] = crc[10] ^ data_sim ^ crc[0];
 assign new_crc[10] = crc[11];
 assign new_crc[11] = crc[12];
 assign new_crc[12] = crc[13];
 assign new_crc[13] = crc[14];
 assign new_crc[14] = crc[15];
-assign new_crc[15] = crc[16] ^ data ^ crc[0];
+assign new_crc[15] = crc[16] ^ data_sim ^ crc[0];
 assign new_crc[16] = crc[17];
 assign new_crc[17] = crc[18];
 assign new_crc[18] = crc[19];
-assign new_crc[19] = crc[20] ^ data ^ crc[0];
-assign new_crc[20] = crc[21] ^ data ^ crc[0];
-assign new_crc[21] = crc[22] ^ data ^ crc[0];
+assign new_crc[19] = crc[20] ^ data_sim ^ crc[0];
+assign new_crc[20] = crc[21] ^ data_sim ^ crc[0];
+assign new_crc[21] = crc[22] ^ data_sim ^ crc[0];
 assign new_crc[22] = crc[23];
-assign new_crc[23] = crc[24] ^ data ^ crc[0];
-assign new_crc[24] = crc[25] ^ data ^ crc[0];
+assign new_crc[23] = crc[24] ^ data_sim ^ crc[0];
+assign new_crc[24] = crc[25] ^ data_sim ^ crc[0];
 assign new_crc[25] = crc[26];
-assign new_crc[26] = crc[27] ^ data ^ crc[0];
-assign new_crc[27] = crc[28] ^ data ^ crc[0];
+assign new_crc[26] = crc[27] ^ data_sim ^ crc[0];
+assign new_crc[27] = crc[28] ^ data_sim ^ crc[0];
 assign new_crc[28] = crc[29];
-assign new_crc[29] = crc[30] ^ data ^ crc[0];
-assign new_crc[30] = crc[31] ^ data ^ crc[0];
-assign new_crc[31] =           data ^ crc[0];
+assign new_crc[29] = crc[30] ^ data_sim ^ crc[0];
+assign new_crc[30] = crc[31] ^ data_sim ^ crc[0];
+assign new_crc[31] =           data_sim ^ crc[0];
 
 always @ (posedge clk or posedge rst)
 begin
