From 888d2d48794828bb4217c185643defb5e751d892 Mon Sep 17 00:00:00 2001
From: Hans Baier <hansfbaier@googlemail.com>
Date: Thu, 17 Oct 2013 12:23:18 +0700
Subject: [PATCH 3/6] remove unused register

---
 simple_spi_top.v |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/simple_spi_top.v b/simple_spi_top.v
index a2bf590..6541e10 100644
--- a/simple_spi_top.v
+++ b/simple_spi_top.v
@@ -95,7 +95,7 @@ module simple_spi_top(
   reg  [7:0]          spcr;       // Serial Peripheral Control   Register ('HC11 naming)
   wire [7:0]          spsr;       // Serial Peripheral Status    Register ('HC11 naming)
   reg  [7:0]          sper;       // Serial Peripheral Extension Register
-  reg  [7:0] treg, rreg; // Transmit/Receive register
+  reg  [7:0]          treg;       // Transmit Register
 
   // fifo signals
   wire [7:0] rfdout;
-- 
1.7.9.5

