From 8ee9798d9e69d5641d928db40388eded005ab301 Mon Sep 17 00:00:00 2001
From: Hans Baier <hansfbaier@googlemail.com>
Date: Thu, 17 Oct 2013 12:30:53 +0700
Subject: [PATCH 6/6] add default to prevent latch creation

---
 simple_spi_top.v |    1 +
 1 file changed, 1 insertion(+)

diff --git a/simple_spi_top.v b/simple_spi_top.v
index b035930..151fe80 100644
--- a/simple_spi_top.v
+++ b/simple_spi_top.v
@@ -154,6 +154,7 @@ module simple_spi #(
        3'b010: dat_o <= rfdout;
        3'b011: dat_o <= sper;
        3'b100: dat_o <= {{ (8-SS_WIDTH){1'b0} }, ss_r};
+      default: dat_o <= 0;
     endcase
 
   // read fifo
-- 
1.7.9.5

