From 34794cdadb2679c076990433c16a7b0b13ecc8c9 Mon Sep 17 00:00:00 2001
From: Stefan Kristiansson <stefan.kristiansson@saunalahti.fi>
Date: Mon, 23 Sep 2013 06:39:51 +0300
Subject: [PATCH 1/2] fix typo in 'wbm_busy' signal

---
 vga_enh_top.v | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vga_enh_top.v b/vga_enh_top.v
index 1f1cb48..5610624 100644
--- a/vga_enh_top.v
+++ b/vga_enh_top.v
@@ -168,7 +168,7 @@ module vga_enh_top (
 
 	// to wb_slave
 	wire stat_avmp, stat_acmp, vmem_swint, clut_swint, hint, vint, sint;
-	wire wmb_busy;
+	wire wbm_busy;
 	reg luint;
 	wire [ 3: 0] cc0_adr_i, cc1_adr_i;
 
-- 
1.8.1.2

